CruzHacks/cruzhacks-2021-website
# Integrate GitHub Actions Pipeline

Problem
=======
As a developer, 
I want to create a CI/CD pipeline 
So we can automate and streamline our code development process 

[Link to Asana Ticket](https://app.asana.com/0/0/1192481576680911/f)

Solution
========
Created a Github Actions CI pipeline. In order to create this pipeline, I had to flesh out the linters, testing suites, and the auto-deploy to production. I also incorporated the `Javascript => Typescript` change as that also had an effect in how we run our linter and the dependencies we have. 

Steps to Verify:
----------------
1. Verify that GithubActions tab runs all 4 tests (skips the `auto-deploy-to-prod`)
      * `static-analysis` - runs linter 
      * `js-tests` - runs jest command 
      * `build` - creates a production build 
      * `auto-deploy-to-prod` - deploys to the [firebase app ](https://cruzhacks-2021-website.web.app/) only in the `main` branch
1. Re-run the workflow to verify that the test all pass 

Screenshots:
-----------------------
<img width="1435" alt="Screen Shot 2020-09-21 at 2 01 25 PM" src="https://user-images.githubusercontent.com/40477553/93821098-f350ef80-fc12-11ea-8b24-72452dd5631c.png">
